ibis/backends/impala/tests/snapshots/test_exprs/test_filter_with_analytic/out.sql

```diff
@@ -1,18 +1,16 @@
-WITH t0 AS (
-  SELECT t4.`col`, NULL AS `filter`
-  FROM `x` t4
-),
-t1 AS (
-  SELECT t0.*
-  FROM t0
-  WHERE t0.`filter` IS NULL
-),
-t2 AS (
-  SELECT t1.`col`, t1.`filter`
-  FROM t1
-)
-SELECT t3.`col`, t3.`analytic`
+SELECT
+  `t2`.`col`,
+  `t2`.`analytic`
 FROM (
-  SELECT t2.`col`, count(1) OVER () AS `analytic`
-  FROM t2
-) t3
\ No newline at end of file
+  SELECT
+    `t1`.`col`,
+    COUNT(*) OVER (ORDER BY NULL ASC NULLS LAST) AS `analytic`
+  FROM (
+    SELECT
+      `t0`.`col`,
+      NULL AS `filter`
+    FROM `x` AS `t0`
+    WHERE
+      NULL IS NULL
```

Comment:
Simplification opportunity here. `filter` is nicely inlined, we could eliminate the whole filter. Not important here of course.